src/main/io/ledstrip.c

```diff
@@ -574,6 +576,77 @@ static void applyLedWarningLayer(bool updateNow, timeUs_t *timer)
     }
 }
 
+#ifdef VTX_COMMON
+static void applyLedVtxLayer(bool updateNow, timeUs_t *timer)
+{
+    static uint16_t frequency = 5800;
+    static uint8_t power = 0;
+    static uint8_t pit = 0;
+    static uint8_t showSettings = false;
+    static uint16_t lastCheck = 0;
+    static bool active = false;
+    static bool blink = false;
+    uint8_t band = 3, channel = 4;
+    uint16_t check = 0;
+
+    if (updateNow) {
+        // keep counter running, so it stays in sync with vtx
+        active = vtxCommonGetBandAndChannel(&band, &channel);
+        vtxCommonGetPowerIndex(&power);
+        vtxCommonGetPitMode(&pit);
+
+        frequency = vtx58frequencyTable[band][channel];
+
+        // check if last vtx values have changed.
+        check = pit + (power << 1) + (band << 4) + (channel << 8);
+        if(!showSettings && check != lastCheck) {
+            // display settings for 3 seconds.
+            showSettings = 15;
+        }
+        lastCheck = check; // quick way to check if any settings changed.
+
+        if(showSettings) {
+            showSettings--;
+        }
+        blink = !blink;
+        *timer += HZ_TO_US(5); // check 5 times a second
+    }
+
+    if(!active) { // no vtx device detected
+        return;
+    }
+
+    hsvColor_t color = { 0, 0, 15};
+    if(showSettings) { // show settings
+        for (int i = 0; i < ledCounts.count && i < 6; ++i) {
+            const ledConfig_t *ledConfig = &ledStripConfig()->ledConfigs[i];
+            if (ledGetOverlayBit(ledConfig, LED_OVERLAY_VTX)) {
+                if(i == 0) {
```

Comment:
you need to count LED_OVERLAY_VTX leds. `i` will work only when no other led modes  are used